### ISSUE TITLE
Review BABE implementation in Gossamer

### ISSUE THREAD

**Issue by lamafab** [w3f/polkadot-spec#227]

**Comment by drskalman** [comment]
@FlorianFranzen aiming to finish this till next week.

**Comment by drskalman** [comment]
A list of 2 things:
- [x] VRF (is schnorr signature implemented in go or is calling into Rust?)
- [x] NextEpochDescriptor.
- [x] Verify SCALE encoding of the digests.
Done by the end of the week.

**Comment by drskalman** [comment]
- interal of VRF review moved to another ticket.
- There has been no major issues just discussion in the channel.
- NextEpochDescriptor is the only thing that is left but need to be specced.

**Comment by drskalman** [comment]
- NextEpochDescriptor is #173 done and look good in Gossamer.
- SCALE encoding and VRF is the remaining point.

**Comment by drskalman** [comment]
Blocked by #46 for now. @FlorianFranzen gets back to review when the everything about the spec is clear.

**Comment by lamafab** [comment]
@FlorianFranzen will review this after #168 is done.

**Comment by FlorianFranzen** [comment]
Wrote up the last few things from my notes in ChainSafe/gossamer#1628 and ChainSafe/gossamer#1629.
